modules/openapi-generator/src/test/resources/3_0/issue_18922.yaml

```diff
@@ -0,0 +1,31 @@
+openapi: 3.0.0
+info:
+  title: API
+  version: 1.0.0
+servers:
+  - url: https://example.com
+paths:
+  /foo:
+    patch:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Bar"
+      responses:
+        "204":
+          description: "No Content"
+components:
+  schemas:
+    Bar:
```

Comment:
what about adding this new schema/model to `modules/openapi-generator/src/test/resources/3_0/go/petstore-with-fake-endpoints-models-for-testing-with-http-signature.yaml` (the existing go test spec) instead and then regenerate the samples?

Reply:
done -  `./bin/generate-samples.sh ./bin/configs/go-petstore.yaml -- -p useOneOfDiscriminatorLookup=true` now reproduces the issue in `model_one_of_with_complex_type.go` (when the mustache changes are reverted).